src_blog/_posts/New-Release-Grain-v0.7-Farro.md

```diff
@@ -0,0 +1,156 @@
+
+let doublePrice = price => price * 2
+
+let metadata = map(property("price") ||> number, doublePrice, metadata)
+// Some(JsonObject([("currency", JsonString("$")), ("price", JsonNumber(238))]))
+```
+
+Deeply nested values can be focused in on with the `proprtyPath` lens. We can even define our own lenses:
+
+```grain
+module Main
+
+from "json" include Json
+use Json.{ type Json }
+use Json.Lenses.{ get, set, propertyPath, (||>), type Lens }
+
+from "result" include Result
+
+let metadata = Result.unwrap(
+  Json.parse("{\"vehicles\":{\"car\":{\"color\":\"red\"}}}")
+)
+
+enum Color {
+  Red,
+  Blue,
+  Green,
+}
+
+// Let's define a lens that works with Color values.
+
+// The getter.
+let colorGet = json => {
+  match (json) {
+    JsonString("red") => Some(Red),
+    JsonString("green") => Some(Green),
+    JsonString("blue") => Some(Blue),
+    _ => None,
+  }
+}
+
+// The setter.
+let colorSet = (newValue, _) => {
+  match (newValue) {
+    Red => Some(JsonString("red")),
+    Green => Some(JsonString("green")),
+    Blue => Some(JsonString("blue")),
+  }
+}
+
+// The lens is just a record with the getter and setter.
+let color = { get: colorGet, set: colorSet }
+
+// Now we can use our new lens!
+
+// Some(Red)
+get(propertyPath(["vehicles", "car", "color"]) ||> color, metadata)
+
+// Returns a new Json value with the color set to "blue"
+set(propertyPath(["vehicles", "car", "color"]) ||> color, Blue, metadata)
+```
+
+As you can see, we can create lenses that can get and set arbitrary values. This is a massive leap forward for the JSON library, and we can't wait to see what you build with it!
+
+## Standard library submodules
+
+Like the new `Json.Lenses` submodule, we're continuing to add new and useful submodules to the standard library, allowing us to add tons of new functionality while keeping the standard library organized. I want to call attention to two in particular—`Char.Ascii` and `List.Associative`.
+
+Grain `Char`s are Unicode scalar values, but there are many properties of the subset of those values that are ASCII that are interesting. The new `Ascii` submodule gives you access to many of those properies, like `Ascii.isAlpha`, without polluting the namespace of the broader `Char` library.
+
+The `List.Associative` submodule provides utilities for working with lists of key-value pairs. This is a fairly common pattern (and is used for JSON objects!), so this submodule finds a welcome home in `List`.
+
+## Grain object files
+
+Previously, the Grain compiler would produce intermediate `.gr.wasm` files when compiling a project. This could be confusing, as these `.wasm`s weren't useful on their own. The compiler now produces `.gro` files, which are better suited to be linked together later, along with reducing the need to completely re-compile a project when some compiler flags change. As a bonus, this also improves compiler performance, resulting in faster compile times.
```

Comment:
Could we mention the 10% number from the [pr](https://github.com/grain-lang/grain/pull/2104), we could also mention that this helps to fix our source map support (though I don't know if it's working fully I was having issues with it when I tried using them for debugging).

Reply:
I don't really want to call out specific numbers because my testing was pretty un-scientific, and I don't think it's worth the time to get solid numbers since it's a side-effect of the thing that changed, and it's not like you're getting 2x better perf or anything.

Reply:
I did add a line about the debugging though.